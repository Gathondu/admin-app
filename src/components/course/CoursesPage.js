import React, {Fragment} from "react";
import PropTypes from "prop-types";
import CourseList from "./CourseList";

const CoursesPage = ({ history }) => {
    return <Fragment>
      <h1 key="title">Courses</h1>
      <input
        key="submit"
        type="submit"
        value="Add Course"
        className="btn btn-primary"
        onClick={() => history.push("/course")}
      />
      <CourseList key="course-list" />
    </Fragment>
  };

CoursesPage.propTypes = {
  history: PropTypes.shape({
    push: PropTypes.func.isRequired
  }).isRequired
};

export default CoursesPage;
