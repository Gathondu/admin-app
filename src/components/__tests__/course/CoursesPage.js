import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import CoursesPage from "../../course/CoursesPage";

describe('CoursesPage Component', () => {
    const setup = () => {
      const props = {
            history: {
                push: jest.fn()
            }
        };

      return {
          props,
          wrapper: shallow(<CoursesPage {...props}/>)
      }
    };
    it('should not regress', () => {
        const {wrapper} = setup();
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    it('should redirect to /course page', () => {
        const {wrapper, props} = setup();
        wrapper.find('input').simulate('click');
        expect(props.history.push).toHaveBeenCalled()
    });
});