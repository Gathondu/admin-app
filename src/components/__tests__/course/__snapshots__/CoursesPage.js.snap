// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoursesPage Component should not regress 1`] = `
<Fragment>
  <h1
    key="title"
  >
    Courses
  </h1>
  <input
    className="btn btn-primary"
    key="submit"
    onClick={[Function]}
    type="submit"
    value="Add Course"
  />
  <Connect(CourseList)
    key="course-list"
  />
</Fragment>
`;
